server.js

```diff
@@ -1,33 +1,142 @@
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import express, { response } from 'express';
+import bodyParser, { json } from 'body-parser';
+import cors from 'cors';
+
+// Data I want to use to create end points with
+import booksData from './data/books.json';
+
+//Instance of express
+const app = express();
 
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+// PORT=9000 npm start
+const port = process.env.PORT || 8081;
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+// Cors makes it easier to use the API, allows API's to say where the requests come from.
+// bodyParser allows express to read json in post requests
+app.use(cors());
+app.use(bodyParser.json());
+
+// Welcome message when you enter the url
+app.get('/', (request, response) => {
+  response.json("Welcome to Claire's book API 🌼 For documentation go to https://books-deployment.herokuapp.com/documentation");
+});
+
+/* --- First endpoint ---
+Endpoint that returns whole books array http://localhost:8081/books
+
+or different data depending on the query string parameter e.g. author, title, language, average rating and page. 
+
+Example routes for author, title, language and average rating:
+http://localhost:8081/books?author=rowling
+http://localhost:8081/books?title=harry
+http://localhost:8081/books?language=eng
+eng, en-GB, en-US, spa, fre, ger, ara, por, grc, mul
+http://localhost:8081/books?averagerating=4
+
+For the the page query a list of 50 results will be returned from the books array that have been pre-sliced using the slice method. 
+What data is returned depends on what number the user has queried in the endpoint e.g. http://localhost:8081/books?page=1. 
+The number specified is stored in the page query variable and will have a 1 subtracted from it and * by 50 to create the initial index number to be used in the slice. This is stored in the startIndex variable and then used in slice to initiate which book should be returned at the beginning of the slice and which should be return at the end, although the end element will be omitted as this is a side effect of the slice method. This will then return 50 books depending upon which page number is written. If the user types a page number that doesn't exist then an empty array is returned.
+*/
+
+app.get('/books', (request, response) => {
+  const { author, title, language, averagerating, page } = request.query;
+
+  if (author) {
```

Comment:
Great with so much filtering ⭐ A suggestion would be to declare a variable and reassign its value to let the user search for more than one thing. Something like:
```
let filteredBooks
if (author) {
  filteredBooks  = //filtered on author
}
if (language) {
  filteredBooks = //filtered on language
}
res.json(filteredBooks)

```